terraform/constellation-cluster/variables.tf

```diff
@@ -100,3 +123,15 @@ variable "microservice_version" {
   type        = string
   description = "Microservice version."
 }
+
+variable "debug" {
+  type        = bool
+  default     = false
+  description = "DON'T USE IN PRODUCTION: Enable debug mode and allow the use of debug images."
+}
+
+variable "create_maa" {
```

Comment:
nit: should be part of azure config

Reply:
I've tried checking if `maaURL` in `azure_config` is set. Unfortunately, we cannot access any field on `azure_config` in the `count` attribute, it throws the error:
```console
The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the resources that the count depends on.
```

I couldn't figure out the exact path the dependencies are resolved in, or, to be more exact, I couldn't find the cause for the error if the dependencies are resolved in the order they should (imo).
With the release in mind I decided not to spend more time on this

Reply:
you could add this to the resource azure_config (which seems better imo to encapsulate azure config logic in one resource) and make an if check.